lib/web_ui/dev/felt

```diff
@@ -56,9 +56,43 @@ install_deps() {
 KERNEL_NAME=`uname`
 if [[ $KERNEL_NAME == *"Darwin"* ]]
 then
-  echo "Running on MacOS. Increase the user limits"
-  ulimit -n 50000
-  ulimit -u 4096
+  echo "Running on MacOS. Will check the file and user limits."
+  # Disable exit if the commands fails. We want to give more actionable
+  # error message.
+  set +e
+  ULIMIT_FILES=`ulimit -n`
+  # Increase the file limit if it is low. Note that these limits are changed
+  # only for this script (for this shell). After felt execution is completed
+  # no change is required to reset the original shell.
+  if [[ $ULIMIT_FILES < 50000 ]]
+  then
+    echo "File limits too low increasing the file limits"
+    # Get the max file limit.
+    MAX_ULIMIT_FILES=`launchctl limit maxfiles | sed -e 's/^[[:space:]]*//' | sed 's/ \{1,\}/ /g' | cut -d' ' -f2`
```

Comment:
I've gotten similar output with `launchctl limit maxfiles | cut -d' ' -f5` (but I'm not sure about the output of this command in different systems, or with different values. -f5 sounds... arbitrary)

Reply:
:D yes! I added the sed -e 's/^[[:space:]]*//' | sed 's/ \{1,\}/ /g' part just to reduce from f5 to f2